### PR TITLE
Improve null validation

### PR DESCRIPTION
**Pull request by AlexanderSher**
This PR restores null validation in method signatures, simplifies null checks and separates parameter validation methods for LLC and HLC to prepare extension method for more complicated validations

### CODE REVIEW

**Comment by AlexanderSher** [comment]
Change isn't required for RC1, but it is desired.

File: samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs [hunk 1]
```diff
@@ -40,14 +40,8 @@ protected DocumentTranslationClient()
...
         {
-            if (endpoint == null)
-            {
-                throw new ArgumentNullException(nameof(endpoint));
-            }
-            if (credential == null)
-            {
-                throw new ArgumentNullException(nameof(credential));
-            }
+            endpoint = endpoint ?? throw new ArgumentNullException(nameof(endpoint));
```
Comment (annelo-msft) [inline samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs:43]: Would it make sense to use Argument.AssertNotNull here instead?  (See: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs#L62)

**Review by annelo-msft — APPROVED** [review]
This is nice - thanks!

**Comment by AlexanderSher** [comment]
Closes https://github.com/Azure/autorest.csharp/issues/1697

**Pull request merged**